lib/hsa/mcwamp_hsa.cpp

```diff
@@ -5241,6 +5247,10 @@ hsa_status_t HSACopy::hcc_memory_async_copy(Kalmar::hcCommandKind copyKind, cons
     // HSA memory copy requires a system-scope acquire before the next kernel command - set flag here so we remember:
     hsaQueue()->setNextKernelNeedsSysAcquire(true);
 
+    // If it's a device to device, the next system sync will need a release
+    if (copyKind ==  Kalmar::hcMemcpyDeviceToDevice)
+      hsaQueue()->setNextSyncNeedsSysRelease(true);
```

Comment:
@scchan if i understand correctly, this corresponds to the first item in the description:

```
after a device to device memory copy, signal that the next sync needs a system release
```

Reply:
yes, that's correct